src/Utilities/QuadraticInterpolator.cc

```diff
@@ -62,37 +51,42 @@ QuadraticInterpolator::initialize(double xmin,
   double x0, x1, x2;
   EMatrix A;
   EVector X, B;
-  for (auto i0 = 0u; i0 <= mN1; ++i0) {
-    x0 = xmin + i0*mXstep;
-    x1 = x0 + 0.5*mXstep;
-    x2 = x0 + mXstep;
+  for (auto i0 = 0u; i0 <= N1; ++i0) {
+    x0 = xmin + i0*xstep;
+    x1 = x0 + 0.5*xstep;
+    x2 = x0 + xstep;
     A << 1.0, x0, x0*x0,
          1.0, x1, x1*x1,
          1.0, x2, x2*x2;
     B << yvals[2u*i0], yvals[2u*i0 + 1u], yvals[2u*i0 + 2u];
     X = A.inverse()*B;
-    mcoeffs[3*i0     ] = X(0);
-    mcoeffs[3*i0 + 1u] = X(1);
-    mcoeffs[3*i0 + 2u] = X(2);
+    vals[3*i0     ] = X(0);
+    vals[3*i0 + 1u] = X(1);
+    vals[3*i0 + 2u] = X(2);
   }
+#ifdef SPHERAL_GPU_ENABLED
+  auto allocator = rm.getAllocator("DEVICE");
+  mDeviceCoeffs = static_cast<double*>(allocator.allocate(N*sizeof(double)));
+  rm.copy(mDeviceCoeffs, vals, N*sizeof(double));
+#endif
+  mcoeffs = vals;
+  mN1 = N1;
+  mXmin = xmin;
+  mXmax = xmax;
+  mXstep = xstep;
 }
 
 //------------------------------------------------------------------------------
 // Destructor
 //------------------------------------------------------------------------------
 QuadraticInterpolator::~QuadraticInterpolator() {
-}
-
-//------------------------------------------------------------------------------
-// Equivalence
-//------------------------------------------------------------------------------
-bool
-QuadraticInterpolator::
-operator==(const QuadraticInterpolator& rhs) const {
-  return ((mN1 == rhs.mN1) and
-          (mXmin == rhs.mXmin) and
-          (mXmax == rhs.mXmax) and
-          (mcoeffs == rhs.mcoeffs));
+  auto& rm = umpire::ResourceManager::getInstance();
+#ifdef SPHERAL_GPU_ENABLED
+  auto allocator = rm.getAllocator("DEVICE");
+  allocator.deallocate(mDeviceCoeffs);
+#endif
+  auto host_allocator = rm.getAllocator("HOST");
+  host_allocator.deallocate(mcoeffs);
```

Comment:
Same with the destructors, unnecessary deallocation calls could be unnecessarily blocking the device